### PR TITLE
all: clean up eigenproblems-related types and consts 

### PR DESCRIPTION
**Pull request by vladimir-ch**
First part of clean up for byte types and consts related to eigenvalue solvers. Review on per-commit basis.

Updates #163

### CODE REVIEW

**Comment by vladimir-ch** [comment]
I'll fix the call to Syev in mat64 if and when this gets a LGTM.

File: lapack.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ type Float64 interface {
 	Dormlq(side blas.Side, trans blas.Transpose, m, n, k int, a []float64, lda int, tau, c []float64, ldc int, work []float64, lwork int)
 	Dpocon(uplo blas.Uplo, n int, a []float64, lda int, anorm float64, work []float64, iwork []int) float64
 	Dpotrf(ul blas.Uplo, n int, a []float64, lda int) (ok bool)
-	Dsyev(jobz EigComp, uplo blas.Uplo, n int, a []float64, lda int, w, work []float64, lwork int) (ok bool)
+	Dsyev(jobz JobEV, uplo blas.Uplo, n int, a []float64, lda int, w, work []float64, lwork int) (ok bool)
```
Comment (btracey) [inline lapack.go]: We have a lot of functions here.

File: lapack.go [hunk 2]
```diff
@@ -122,6 +122,12 @@ const (
 	EigBoth EigComp = 'I'
 )
 
+// JobEV specifies whether eigenvectors will be computed in Dsyev.
```
Comment (btracey) [inline lapack.go]: I think these all would be better as `EVJob` , `EVCompute`, `EVSide`, etc.. It matches both `SVDJob` and `EigDecomp`.

Comment (vladimir-ch) [inline lapack.go]: I changed `SideEV` to `EVSide` and `JobEV` to `EVJob`. The rest I'll do in a next PR, I don't want this one to get too messy.

**Comment by vladimir-ch** [comment]
PTAL

**Comment by btracey** [comment]
LGTM

**Pull request merged**